### PR TITLE
Fix Vnmls_S fast path (F64: losing input d value). Fix Vnmla_S & Vnmls_S slow paths (using fused inst.s). Fix Vfma_V slow path not using StandardFPSCRValue().

### PR DESCRIPTION
**Pull request by LDj3SNuD**
Add Vfms_V inst. with Fma fast path and slow path.

Add Vfma_S & Vfms_S Fma fast paths.
Add Vfnma_S & Vfnms_S Sse fast paths.

Add Tests for all affected inst.s.

Nits.

### CODE REVIEW

File: ARMeilleure/Instructions/InstEmitSimdArithmetic32.cs [hunk 1]
```diff
@@ -415,24 +444,7 @@ public static void Vnmls_S(ArmEmitterContext context)
...
-                        Operand mask = X86GetScalar(context, -0f);
-                        d = context.AddIntrinsic(Intrinsic.X86Xorps, mask, d);
-                        return context.AddIntrinsic(Intrinsic.X86Addss, d, res);
-
-                    }
-                    else
-                    {
-                        Operand res = context.AddIntrinsic(Intrinsic.X86Mulsd, n, m);
-                        Operand mask = X86GetScalar(context, -0d);
-                        d = context.AddIntrinsic(Intrinsic.X86Xorpd, mask, res);
```
Comment (LDj3SNuD) [inline ARMeilleure/Instructions/InstEmitSimdArithmetic32.cs:432]: Fix Vnmls_S fast path (F64: losing input d value).

File: ARMeilleure/Instructions/InstEmitSimdArithmetic32.cs [hunk 2]
```diff
@@ -375,36 +419,21 @@ public static void Vnmla_S(ArmEmitterContext context)
...
                 {
-                    return context.Negate(context.Add(op1, context.Multiply(op2, op3)));
+                    return context.Subtract(context.Negate(op1), context.Multiply(op2, op3));
                 });
             }
             else
             {
                 EmitScalarTernaryOpF32(context, (op1, op2, op3) =>
                 {
-                    return EmitSoftFloatCall(context, nameof(SoftFloat32.FPNegMulAdd), op1, op2, op3);
```
Comment (LDj3SNuD) [inline ARMeilleure/Instructions/InstEmitSimdArithmetic32.cs:407]: Fix Vnmla_S & Vnmls_S slow paths (using fused inst.s).

File: ARMeilleure/Instructions/InstEmitSimdArithmetic32.cs [hunk 3]
```diff
@@ -445,7 +457,8 @@ public static void Vnmls_S(ArmEmitterContext context)
             {
                 EmitScalarTernaryOpF32(context, (op1, op2, op3) =>
                 {
-                    return EmitSoftFloatCall(context, nameof(SoftFloat32.FPNegMulSub), op1, op2, op3);
```
Comment (LDj3SNuD) [inline ARMeilleure/Instructions/InstEmitSimdArithmetic32.cs:448]: Fix Vnmla_S & Vnmls_S slow paths (using fused inst.s).

**Review by gdkchan — APPROVED** [review]
lgtm, thanks. I noticed that the *PD FMA instructions were removed though, any reason for that?

**Comment by LDj3SNuD** [comment]
>  I noticed that the *PD FMA instructions were removed though, any reason for that?

Because in A32 there are only 4 advanced fused type instructions, the scalar ones we already have (Vfma_S, Vfms_S, Vfnma_S and Vfnms_S type F32 and F64) and the vector ones Vfma_V and Vfms_V type F32 only (so the 2 intrinsic _ps should be used for these vector variants).

File: ARMeilleure/Instructions/InstEmitSimdArithmetic32.cs [hunk 4]
```diff
@@ -256,24 +256,25 @@ public static void Vext(ArmEmitterContext context)
...
-                // The 64-bit variant will never be used.
-                EmitVectorTernaryOpF32(context, Intrinsic.X86Vfmadd231ps, Intrinsic.X86Vfmadd231pd);
+                EmitVectorTernaryOpF32(context, Intrinsic.X86Vfmadd231ps);
             }
             else
             {
                 EmitVectorTernaryOpF32(context, (op1, op2, op3) =>
                 {
-                    return EmitSoftFloatCall(context, nameof(SoftFloat32.FPMulAdd), op1, op2, op3);
+                    return EmitSoftFloatCallDefaultFpscr(context, nameof(SoftFloat32.FPMulAddFpscr), op1, op2, op3);
```
Comment (LDj3SNuD) [inline ARMeilleure/Instructions/InstEmitSimdArithmetic32.cs:265]: Fix Vfma_V slow path not using StandardFPSCRValue().

**Comment by LDj3SNuD** [comment]
Ready to be reviewed.

**Review by riperiperi — APPROVED** [review]
LGTM. Sorry for the delay. I haven't had a chance to test this, but it all looks good.

**Pull request merged**